Login, forgotten password and (minor) registration adjustments

marcusxss opened the pull request:
Opening this up as a draft to begin with, both for feedback, and because I need to test it a bit further.

I'm also not sure if this should target `develop` or `rc`...?
At least I think someone else should help verify nothing is broken if it should target `rc`.

Another question, also relevant to the target branch, is how likely this is to break the login/home screen theme changes that some instances do.

# Significant code changes

- `field_input` and `field_password` are expanded with the possibility to set a placeholder text - for both Frio and the default templates.
- Frio's `field_input` already has a way to disable the label by setting label to false. That functionality is copied to `field_password` for Frio, as well as the general `field_input` and `field_password` templates.

# UX / Visual changes

Visually it doesn't make *that* many changes, but I did have to change the code quite a lot comparatively, so that's why there's some potential for breakage. Fx. the username and password field are no longer generated by the `field` templates but instead input elements are added directly to the template to get more control over them, while not affecting everything else on Friendica.

The main changes are:
- Login/Logout -> Sign in/Sign out
- Autofocus is added in a few places where it was missing, so it preselects the first input element on the login and registration forms, and preselects the nickname/email field in the forgotten password form.
- A larger login action button, more similar to the register action
- Separate the register action a bit from the rest of the form
- Placeholder texts rather than labels
- The OpenID label is now more of a header of sorts. But not an HTML header, because it's in the main login form is both above and below it. It's just for visual separation.
- Increase contrast between the background and text on the register button

It's quite inspired by Facebook's login prompt, which it already seemed to be to begin with.

# Screenshots

## Navigation

Text is added to the sign in action. The hover text is removed.

### Before

<img width="156" height="56" alt="image" src="https://github.com/user-attachments/assets/e66eb966-7c92-420d-b1a5-0de4636c6cd9" />

### After

<img width="173" height="52" alt="image" src="https://github.com/user-attachments/assets/f7429bb7-e971-46b3-958b-05a61629f1ad" />

## Home

### Before

<img width="759" height="601" alt="image" src="https://github.com/user-attachments/assets/4047331e-e7a5-460d-8f84-8c403637c6f5" />

### After

<img width="661" height="732" alt="image" src="https://github.com/user-attachments/assets/c30d397d-2a64-43c4-8f4c-d8bc9df0a7d5" />

Smaller resolution:
<img width="319" height="619" alt="image" src="https://github.com/user-attachments/assets/61b257d2-d96b-4289-bf8c-b6639da23393" />

## /login

### Before

<img width="694" height="407" alt="image" src="https://github.com/user-attachments/assets/48ace081-ae1d-4d40-9f6a-d1aedd1ed5aa" />

### After

<img width="697" height="565" alt="image" src="https://github.com/user-attachments/assets/e314c613-4c42-47ea-9903-1bf6fa14ca34" />

I think, as mentioned in #15215, that at some point we should remove /login and make it redirect to Home.

## Modal

### Before

<img width="666" height="432" alt="image" src="https://github.com/user-attachments/assets/fd412ed1-d354-40ed-acb2-f4b13ef0a8db" />

### After

<img width="671" height="644" alt="image" src="https://github.com/user-attachments/assets/36394cd9-575f-4557-aebc-6704ad54eb94" />

## Forgotten password

Only adjusted it a little

### Before

<img width="1181" height="313" alt="image" src="https://github.com/user-attachments/assets/0ff95419-3b57-4b44-a0d6-a8fda5d5a39f" />

### After

<img width="655" height="340" alt="image" src="https://github.com/user-attachments/assets/649e52c9-789e-4f7c-9697-7055a85f5fe4" />

# Vier

## Home - Before

<img width="564" height="490" alt="image" src="https://github.com/user-attachments/assets/91a76e89-d71c-49c0-9fc4-2f7c63eb6dae" />

## Home - After

<img width="564" height="490" alt="image" src="https://github.com/user-attachments/assets/401866a4-08b9-4331-95d2-266d4980dc42" />

marcusxss committed:
Log in screen adjustments

nupplaphil commented:
Can you check if the accessibility with voice over is still working in case we don't use labels?

Just reading the name and the password won't be enough in case it is set, as the user won't know if it is in the correct field.

marcusxss commented:
Yes, very good point. If it breaks and there's no easy adjustments available, alternatively there could still be labels that are just `sr-only`.

Do we have any documentation of how this is usually tested? I could try Orca.

foss- commented:
A few thoughts since you are touching this dialog:

- would `Create your account` be better than `Sign up now`
- `>>` not needed for that button text
- Would be nice to have a big headline for this dialog like `Sign in` (or Log in)
- when using Sign in, also adapt Login button text
- can text be added in the middle of the divider line above the `Sign up now` button like `New to Friendica?` - stolen from amazon, but it is a nice touch:
<img src="https://github.com/user-attachments/assets/59d7b835-736e-4b9a-bab1-66a131aa1132" width="350">

marcusxss commented:
Regarding these:
> would Create your account be better than Sign up now
>` >>` not needed for that button text

...the code (source string) does not say "Sign up now" or add the ">>" in front. The code says "Register", and then the English translation has changed that into "Sign up now >>".

...but if it *were* changed to "Create new account" this would break existing translations, and so it would no longer be translated to "Sign up now >>", at least for a while. Translators could of course then also translate the new text to "Sign up now >>", or anything else. Not sure who's making decisions on English translations on Transifex.

marcusxss commented:
I originally preferred "Register" but maybe "Create account" is also clearer?

I think that text in the middle of the line does not look that great and it's also a little annoying to write the logic for.

...but here's a new version with some of the suggestions. I think it's better, too. The header in particular:
<img width="554" height="706" alt="image" src="https://github.com/user-attachments/assets/9f6d91b7-506c-4304-a0bf-b2a4f709d227" />

foss- commented:
This looks much more mature - thank you.

marcusxss committed:
Additional log in screen adjustments

marcusxss commented:
Alright, I did a screen reader test before and after using `Orca`, and it appeared to be the same experience as currently.
It could read the field names whether filled or not in both cases.

I'm gonna do some final testing and possible adjustments and will then mark it as ready for review.

marcusxss committed:
Login screen: Go back to using field_input and field_password, add sign in text

marcusxss commented:
Updated the MR contents, the description and screenshots.

Moved away from using a custom form on the login page to set placeholder text, and instead made the regular field_input/field_password fields support setting it, so it can be reused everywhere.
